eq_cir_proxy_service/services/instrument/instrument_conversion_service.py

```diff
@@ -0,0 +1,115 @@
+"""This module requests conversion of the instrument from Converter Service."""
+
+import os
+
+from fastapi import HTTPException, status
+from httpx import AsyncClient, RequestError
+from semver import Version
+from structlog import get_logger
+
+from eq_cir_proxy_service.exceptions import exception_messages
+from eq_cir_proxy_service.types.custom_types import Instrument
+
+logger = get_logger()
+
+
+def safe_parse(source: str, version: str) -> Version:
+    """Safely parses a version string into a Version object."""
+    try:
+        return Version.parse(version)
+    except ValueError as e:
+        # Logs full traceback with context
+        logger.exception("Error parsing version:", source=source, version=version)
+        # Client sees only this clean message
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail={
+                "status": "error",
+                "message": f"Invalid {source} version: {version}",
+            },
+        ) from e
+
+
+async def convert_instrument(instrument: Instrument, target_version: str) -> Instrument:
+    """Requests conversion of the instrument from Converter Service.
+
+    Parameters:
+    - instrument: The instrument.
+    - current_version: The current version of the instrument.
+    - target_version: The target version of the instrument.
+
+    Returns:
+    - dict: The converted instrument.
+    """
+    if not instrument.get("validator_version"):
+        logger.error("Instrument version is missing")
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail={"status": "error", "message": exception_messages.EXCEPTION_400_INVALID_INSTRUMENT},
+        )
+
+    current_version = str(instrument["validator_version"])
+    parsed_current_version = safe_parse("current", current_version)
+    parsed_target_version = safe_parse("target", target_version)
+
+    if parsed_current_version < parsed_target_version:
+        logger.info("Instrument requires updating")
```

Comment:
Minor, but do we need three separate log lines here with no lines of code in between? Could just be one?